### ISSUE TITLE
Startup crash under 4.9.0 beta with email-to-SMS gateway message

### ISSUE THREAD

**Issue by strugee** [WhisperSystems/Signal-Android#6841]
I have:
- [x ] searched open and closed issues for duplicates
- [x ] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description

The newest beta is crashing immediately on startup. Based on the stack trace, it appears that it's due to some messages I have from NASA's [Spot the Station][] service, which creates messages with the sender set to an email address.

Here's the stack trace and a couple lines before it for context:

```
08-01 13:02:50.094  4752  7122 I ActivityManager: Start proc 11609:org.thoughtcrime.securesms/u0a105 for activity org.thoughtcrime.securesms/.RoutingActivity
08-01 13:02:50.211 11609 11609 I MultiDex: VM with version 2.1.0 has multidex support
08-01 13:02:50.212 11609 11609 I MultiDex: install
08-01 13:02:50.212 11609 11609 I MultiDex: VM has multidex support, MultiDex support library is disabled.
08-01 13:02:50.215 11609 11609 W PartProvider: onCreate()
08-01 13:02:50.261 11609 11609 W PersistentAlarmManagerListener: class org.thoughtcrime.securesms.service.RotateSignedPreKeyListener scheduling for: 1501615220840
08-01 13:02:50.266 11609 11609 W PersistentAlarmManagerListener: class org.thoughtcrime.securesms.service.DirectoryRefreshListener scheduling for: 1501620162997
08-01 13:02:50.314 11609 11609 D JVM     : JVM::Initialize@[tid=11609]
08-01 13:02:50.314 11609 11609 D JVM     : JVM::JVM@[tid=11609]
08-01 13:02:50.315 11609 11609 D JVM     : LoadClasses
08-01 13:02:50.315 11609 11609 D JVM     : name: org/webrtc/voiceengine/BuildInfo
08-01 13:02:50.315 11609 11609 D JVM     : name: org/webrtc/voiceengine/WebRtcAudioManager
08-01 13:02:50.315 11609 11609 D JVM     : name: org/webrtc/voiceengine/WebRtcAudioRecord
08-01 13:02:50.315 11609 11609 D JVM     : name: org/webrtc/voiceengine/WebRtcAudioTrack
08-01 13:02:50.349 11609 11628 E AndroidRuntime: FATAL EXCEPTION: JobConsumer-4
08-01 13:02:50.349 11609 11628 E AndroidRuntime: Process: org.thoughtcrime.securesms, PID: 11609
08-01 13:02:50.349 11609 11628 E AndroidRuntime: Theme: themes:{default=overlay:system, iconPack:system, fontPkg:system, com.android.systemui=overlay:system, com.android.systemui.navbar=overlay:system}
08-01 13:02:50.349 11609 11628 E AndroidRuntime: java.lang.AssertionError: Not e164: HQ-spotthestation@mail.nasa.gov
08-01 13:02:50.349 11609 11628 E AndroidRuntime: 	at org.thoughtcrime.securesms.database.Address.toPhoneString(Address.java:112)
08-01 13:02:50.349 11609 11628 E AndroidRuntime: 	at org.thoughtcrime.securesms.recipients.RecipientProvider.getIndividualRecipientDetails(RecipientProvider.java:136)
08-01 13:02:50.349 11609 11628 E AndroidRuntime: 	at org.thoughtcrime.securesms.recipients.RecipientProvider.getRecipientDetailsSync(RecipientProvider.java:130)
08-01 13:02:50.349 11609 11628 E AndroidRuntime: 	at org.thoughtcrime.securesms.recipients.RecipientProvider.getRecipient(RecipientProvider.java:83)
08-01 13:02:50.349 11609 11628 E AndroidRuntime: 	at org.thoughtcrime.securesms.recipients.RecipientProvider.getRecipients(RecipientProvider.java:99)
08-01 13:02:50.349 11609 11628 E AndroidRuntime: 	at org.thoughtcrime.securesms.recipients.RecipientFactory.getRecipientsFor(RecipientFactory.java:53)
08-01 13:02:50.349 11609 11628 E AndroidRuntime: 	at org.thoughtcrime.securesms.jobs.MmsReceiveJob.isBlocked(MmsReceiveJob.java:89)
08-01 13:02:50.349 11609 11628 E AndroidRuntime: 	at org.thoughtcrime.securesms.jobs.MmsReceiveJob.onRun(MmsReceiveJob.java:60)
08-01 13:02:50.349 11609 11628 E AndroidRuntime: 	at org.whispersystems.jobqueue.JobConsumer.runJob(JobConsumer.java:76)
08-01 13:02:50.349 11609 11628 E AndroidRuntime: 	at org.whispersystems.jobqueue.JobConsumer.run(JobConsumer.java:46)
08-01 13:02:50.351  4752  4837 W ActivityManager:   Force finishing activity org.thoughtcrime.securesms/.RoutingActivity
```

There are actually more Signal-related logs below, but I'm guessing this is all you need - let me know if that's not the case and I can post the other ones too.

 [Spot the Station]: https://spotthestation.nasa.gov/

### Device info

**Device:** Nexus 6P
**Android version:** CyanogenMod 13.0-20161220-SNAPSHOT-ZNH5YAO3XH-angler based on Android 6.0.1
**Signal version:** 4.9.0 beta

**Comment by ruicruzpt** [comment]
Ongoing information in this thread: https://github.com/WhisperSystems/Signal-Android/issues/6838

**Comment by strugee** [comment]
@ruicruzpt right, should have specified - I read that issue and based on the stack traces I think this is a different bug than what people are seeing in that issue. That's why I filed separately.

**Comment by moxie0** [comment]
Thanks! I think this issue should be fixed with 5f9a3b7930070d411ea53fb09e69fadc296a9558. I'll publish a new beta with the fix once I take care of the crash in #6838.

**Comment by strugee** [comment]
Thanks so much @moxie0! Can confirm this is fixed as of 4.9.2.
